test/sweet_with_no_egg_use_case

hazemka opened the pull request:
- edit on findSweetsFreeEggs function to make it throw Exception when not found sweets
- add function to create recipe for sweet with no egg on CreateRecipeHelper.kt

hazemka committed:
make unit test for SweetWithNoEggsUseCase.kt
- edit on findSweetsFreeEggs function to make it throw Exception when not found sweets
- add function to create recipe for sweet with no egg on CreateRecipeHelper.kt

skaik-mo reviewed (changes requested):
- Replace non-null assertions (!!) with safe calls (?.) or  OrEmpty.
- Use assertEquals() instead of assertTrue() in tests for clearer failure messages and better readability.
- Add default null values to Recipe class properties instead of relying on CreateRecipeHelper for null safety.

hazemka committed:
Merge branch 'develop' into test/sweet_with_no_egg_use_case

hazemka committed:
edit some test cases and improve use case for SweetWithNoEggsUseCase.kt
- replace force unwrap with null safety
- change filter place to make class more testable
- change assert true to assert equals for more readability

hazemka committed:
Merge remote-tracking branch 'origin/test/sweet_with_no_egg_use_case' into test/sweet_with_no_egg_use_case

skaik-mo reviewed (approved):
Great Job, Keep it up Hazem

BassantM committed:
delete failed and unused test case also it is fully covered

BassantM reviewed (approved):
Good job.
